earth_enterprise/src/fusion/dbmanifest/dbmanifest.cpp

```diff
@@ -327,6 +332,64 @@ void DbManifest::GetManifest(geFilePool &file_pool,
   GetXmlManifest(stream_manifest);
 }
 
+// Read POI file and extract the data file names from within it and then add
+// them to the manifest too
+void DbManifest::GetPoiDataFiles(const std::string& poi_file,
+                     std::vector<ManifestEntry>* stream_manifest,
+                     std::vector<ManifestEntry>* search_manifest)
+{
+  notify(NFY_DEBUG,
+        "Parsing POI file %s looking for data files", poi_file.c_str());
+  size_t stream_poi_file_idx = stream_manifest->size() - 1;
+  size_t search_poi_file_idx = search_manifest->size() - 1;
+  khxml::DOMLSParser *parser = CreateDOMParser();
+  if (parser) {
+    khxml::DOMDocument *doc = ReadDocument(parser, poi_file);
+    if (doc) {
+      try {
+        if (khxml::DOMElement *root = doc->getDocumentElement()) {
+          if (khxml::DOMElement* el_search_table = GetFirstNamedChild(root, "SearchTableValues")) {
+            std::vector<std::string> data_files;
+            FromElementWithChildName(el_search_table, "SearchDataFile", data_files);
+            if (data_files.empty()) {
+              notify(NFY_WARN,
+                    "No data files defined for POI file %s", poi_file.c_str());
+            }
+            for (uint i = 0; i < data_files.size(); ++i) {
+
+              const std::string& data_file = data_files[i];
+              
+              // poi data files have to be set aside until after AddDB processing is done
+              // so we hide them in a special location now within the manifest and will
+              // use them when we crate an 'upload' manifest during DBsync processing
+              if (stream_manifest != search_manifest) {
+                notify(NFY_DEBUG, "Adding POI datafile as a dependent file to stream manifest: %s", data_file.c_str());
+                stream_manifest->at(stream_poi_file_idx).dependents.push_back(data_file);
+              }
+              search_manifest->at(search_poi_file_idx).dependents.push_back(data_file);
```

Comment:
Very minor: you've got adding the file to the vector and printing the notification in opposite orders here and a couple of lines above.

Reply:
Good catch.  I should be consistent.